contracts/token/erc721/extensions/ERC721HybridMinting.sol

```diff
@@ -0,0 +1,226 @@
+        address to;
+        uint256 quantity;
+    }
+
+    function _mintByQuantity(address to, uint256 quantity) internal  {
+        ERC721Psi._mint(to, quantity);
+    }
+
+    function _batchMintByQuantity(Mint[] memory mints) internal  {
+        for (uint i = 0; i < mints.length; i++) {
+            Mint memory m = mints[i];
+            _mintByQuantity(m.to, m.quantity);
+        }
+    }
+
+    function _mintByID(address to, uint256 tokenId) internal {
+        require(tokenId < bulkMintThreshold(), "must mint below threshold"); 
+        ERC721._mint(to, tokenId);
+        _idMintTotalSupply++;
+    }
+
+    function _batchMintByID(address to, uint256[] memory tokenIds) internal {
+        for (uint i = 0; i < tokenIds.length; i++) {
+            _mintByID(to, tokenIds[i]);
+        }
+    }
+
+    struct IDMint {
+        address to;
+        uint256[] tokenIds;
+    }
+
+    function _batchMintByIDToMultiple(IDMint[] memory mints) internal  {
+        for (uint i = 0; i < mints.length; i++) {
+            IDMint memory m = mints[i];
+            _batchMintByID(m.to, m.tokenIds);
+        }
+    }
+
+    function exists(uint256 tokenId) public view virtual returns (bool) {
+        return _exists(tokenId);
+    }
+
+    // Overwritten functions from ERC721/ERC721Psi with split routing
+
+
+    function _exists(uint256 tokenId) internal view virtual override(ERC721, ERC721PsiBurnable) returns (bool) {
+        if (tokenId < bulkMintThreshold()) {
+            return ERC721._exists(tokenId);
+        }
+        return ERC721PsiBurnable._exists(tokenId);
+    }
+
+    function _transfer(address from, address to, uint256 tokenId) internal virtual override(ERC721, ERC721Psi) {
+        if (tokenId < bulkMintThreshold()) {
+            ERC721._transfer(from, to, tokenId);
+        } else {
+            ERC721Psi._transfer(from, to, tokenId);
+        }
+    }
+
+    function ownerOf(uint256 tokenId) public view virtual override(ERC721, ERC721Psi) returns (address) {
+        if (tokenId < bulkMintThreshold()) {
+            return ERC721.ownerOf(tokenId);
+        }
+        return ERC721Psi.ownerOf(tokenId);
+    }
+
+    function _burn(uint256 tokenId) internal virtual override(ERC721, ERC721PsiBurnable) {
+
+        if (tokenId < bulkMintThreshold()) {
+            ERC721._burn(tokenId);
+            _idMintTotalSupply--;
+        } else {
+            ERC721PsiBurnable._burn(tokenId);
+        }
+    }
+
+    function burn(uint256 tokenId) public virtual {
+        require(_isApprovedOrOwner(_msgSender(), tokenId), "ERC721: caller is not token owner or approved");
+        _burn(tokenId);
+    }
+
+    function burnBatch(uint256[] calldata tokenIDs) external {
+        for (uint i = 0; i < tokenIDs.length; i++) {
+            burn(tokenIDs[i]);
+        }
+    }
+
+    // 
+    function _safeMint(address to, uint256 quantity) internal virtual override(ERC721, ERC721Psi) {
+        return super._safeMint(to, quantity);
```

Comment:
wonder if it makes sense to explicitly override and call `ERC721Psi._safeMint(to, quantity)` here since thats what is actually being called. Same for the two overrides below